### PR TITLE
Add rebase continue and abort, reworked commit view, and more

### PR DESCRIPTION
**Pull request by ten3roberts**
This PR also includes a slight rework in the commit view, allowing it to prepopulate the editor with "Merged branch master with origin/master" which happens if you commit during a merge. The previous behavior was an empty commit message.

Additionally, there is now a rebasing section in the status view, showing the rebasing head, as well as the interactive steps remaining.

I have also fixed auto refreshing of the status view when pulling and rebasing, so that it properly shows the merge/rebase conflicts if merging/pulling/rebasing failed due to a conflict.

Let me know what you think

### CODE REVIEW

**Comment by ten3roberts** [comment]
image
image

Rebase in progress view, and commit during merge view

File: lua/neogit/popups/rebase.lua [hunk 1]
```diff
@@ -10,6 +10,12 @@ function M.create()
   local p = popup
     .builder()
     :name("NeogitRebasePopup")
+    :action("r", "Continue rebase", function()
```
Comment (EpiCanard) [inline lua/neogit/popups/rebase.lua:13]: The binding `c` isn't more intuitive ? c for continue.

File: lua/neogit/popups/rebase.lua [hunk 2]
```diff
@@ -10,6 +10,12 @@ function M.create()
...
+      rebase.continue()
+    end)
+    :action("a", "Abort rebase", function()
```
Comment (EpiCanard) [inline lua/neogit/popups/rebase.lua:16]: Can you add also the option for --skip ?

File: lua/neogit/popups/rebase.lua (hunk 1 above)
Comment (EpiCanard) [reply lua/neogit/popups/rebase.lua:13]: You can use `action_if` to only display these actions when there is rebase in progress ?

Comment (ten3roberts) [reply lua/neogit/popups/rebase.lua:13]: I used `rr` as that is the binding in vim-fugitive, partly because it has the same double tap as `cc` for commit.

Comment (EpiCanard) [reply lua/neogit/popups/rebase.lua:13]: I never used vim-fugitive so I was just thinking of a binding more intuitive for any users

Comment (ten3roberts) [reply lua/neogit/popups/rebase.lua]: Alright, I've added action_if and skip as well

Comment (EpiCanard) [reply lua/neogit/popups/rebase.lua]: Thank you for your work !

**Comment by TimUntersberger** [comment]
thank you for your work! I'd suggest keeping the mapping for rebase continue as `r`, since all of the keybindings are ideally a copy of magit (they use r).

**Comment by ten3roberts** [comment]
Right, I will revert that change and use rr, as well as fix the lints.

Do you run selene via lsp in your editor, or only as a CI?

**Comment by TimUntersberger** [comment]
I usually just let the CI handle it.

**Comment by ten3roberts** [comment]
Fixed the unused `get_commit_view` function.

It was marked with a selene allow comment, but it seemed to complain anyway.

**Comment by ten3roberts** [comment]
On another note: while writing this feature I notices the `hl` module was quite messy.

Highlights are defined in two different places, and uses bg color matching to detect if a highlight has been overridden, which doesn't work if you define a highlight by `M.link`, as I initially attempted.

I think a better idea would be to use a simple `highlight default {name} {value}`, which makes it easy for a user to provide their own

**Comment by ten3roberts** [comment]
I discovered that rebase elsewhere didn't refresh, and fixed it.

Other than that, rebasing is now working magically, to the point that I now find myself using tmp commits and actualy squashing them before pushing; something I've never done before.

I would like to contribute more to this project, and if so what is the current issue prioritization?

**Comment by TimUntersberger** [comment]
> Other than that, rebasing is now working magically, to the point that I now find myself using tmp commits and actualy squashing them before pushing; something I've never done before.

That sound awesome!

> I would like to contribute more to this project

Thank you haha

> what is the current issue prioritization?

I haven't really developed anything in a few months, so I don't really have a prioritization. I usually just worked on whatever I needed the most for my own workflow. Right now I'm just overseeing PRs. 

So just work on whatever you need the most yourself. (Though I guess merge conflicts would be a major improvement)

**Comment by aleprovencio** [comment]
Sorry about being a little nosy here!

Since `diffview` recently released it's new "merge tool mode", isn't this "merge conflicts" problem already addressed @TimUntersberger ?

@ten3roberts, I think that a `pre-commit` integration would be really useful, and affects a lot of people. Please, take a look at #339 !

**Comment by TimUntersberger** [comment]
@ten3roberts if you are happy with the PR then I'm going to merge this now. Thanks again!

**Comment by ten3roberts** [comment]
Yes, I am happy with the PR. Looking to pick up other issues such as the pre-commit hook one soon

**Pull request merged**